lib/Conch/Controller/WorkspaceRoom.pm

```diff
@@ -55,15 +53,20 @@ sub replace_rooms ($c) {
 		});
 	}
 
-	if ( $workspace->name eq 'GLOBAL' ) {
+	my $uwr = $c->stash('user')->search_related('user_workspace_roles',
```

Comment:
Unless I'm misreading, we do this query a lot. Should we abstract it?

Reply:
I thought about putting a resultset for it into the stash, but decided to wait and see...

another possibility is writing a bunch of helper subs like 'current_user', 'current_workspace', 'current_device' etc that leveraged the \*_id path arguments. again I wanted to stare at the system a little bit from 1000 feet to see if the highways made better sense somewhere else.

Reply:
If I'm counting right, in this PR, we use that same query six or seven times. I suspect our usage will only grow. So, one way or another, I'd like to abstract it.